hc/api/tests/test_pause.py

```diff
@@ -9,19 +10,26 @@ def test_it_works(self):
         check.save()
 
         url = "/api/v1/checks/%s/pause" % check.code
-        r = self.client.post(url, "", content_type="application/json",
+        resp = self.client.post(url, "", content_type="application/json",
                              HTTP_X_API_KEY="abc")
 
         ### Assert the expected status code and check's status
+        check.refresh_from_db()
+        self.assertEqual(resp.status_code, 200)
+        self.assertEqual(check.status, 'paused')
 
     def test_it_validates_ownership(self):
         check = Check(user=self.bob, status="up")
         check.save()
 
         url = "/api/v1/checks/%s/pause" % check.code
-        r = self.client.post(url, "", content_type="application/json",
+        resp = self.client.post(url, "", content_type="application/json",
                              HTTP_X_API_KEY="abc")
 
-        self.assertEqual(r.status_code, 400)
+        self.assertEqual(resp.status_code, 400)
 
         ### Test that it only allows post requests
+        # Reference -->>> https://docs.djangoproject.com/en/dev/topics/testing/advanced/
+        req_factory = RequestFactory()
+        req = req_factory.post(url, "", content_type="application/json", HTTP_X_API_KEY="abc")
+        self.assertEqual(req.method, 'POST')
```

Comment:
While RequestFactory is definitely good to explore, it isn't exactly needed in this scenario. We want to verify that the API endpoint "only allows POST requests". An easier way to test that would be need to send it non-POST requests with the right parameters and assert that it was rejected/error. Like in most other cases, that can be done with the test client.